### ISSUE TITLE
Add support for x86

### ISSUE THREAD

**Issue by nfrechette** [nfrechette/acl#24]
Appveyor already builds x86 but it has not been tested beyond the unit tests passing.
`acl_compressor.py` needs to be ran on CMU to properly validate with: vs2015, vs2017, gcc5, clang5.

Add x86 support to Travis CI.

**Comment by nfrechette** [comment]
I took the opportunity to run the numbers with all the compilers and various number of cores allocated. I have a desktop with 6 physical cores, 12 logical. I run Ubuntu under a VirtualBox VM with 6 cores allocated. To minimize noise, most runs were made with 3 clips compressed in parallel.

Elapsed time is split into two: the wall clock total time it took to execute things and the cumulative time per thread. The smaller number is the wall clock time. Wall clock time isn't available for Ubuntu due to a bug in the python script but it doesn't matter all that much.

TL;DR: x86 sucks, clang5 rocks, vs2017 is a tiny bit better than vs2015

CMU, x86, vs2015, avx, 11 cores
Compressed 2534 clips in 00h 15m 52.91s
Compressed 67.04 MB, Elapsed 02h 39m 36.29s, Ratio [21.32 : 1], Max error [0.0479]
CMU, x64, vs2015, avx, 11 cores
Compressed 2534 clips in 00h 09m 23.07s
Compressed 67.04 MB, Elapsed 01h 35m 51.23s, Ratio [21.32 : 1], Max error [0.0479]

CMU, x86, vs2017, avx, 11 cores
Compressed 2534 clips in 00h 14m 32.37s
Compressed 67.04 MB, Elapsed 02h 31m 14.03s, Ratio [21.32 : 1], Max error [0.0479]
CMU, x86, vs2017, avx, 3 cores
Compressed 2534 clips in 00h 32m 18.88s
Compressed 67.04 MB, Elapsed 01h 32m 00.76s, Ratio [21.32 : 1], Max error [0.0479]

CMU, x64, vs2017, avx, 11 cores
Compressed 2534 clips in 00h 09m 02.51s
Compressed 67.04 MB, Elapsed 01h 32m 12.35s, Ratio [21.32 : 1], Max error [0.0479]
CMU, x64, vs2017, avx, 6 cores
Compressed 2534 clips in 00h 09m 48.78s
Compressed 67.04 MB, Elapsed 00h 54m 16.00s, Ratio [21.32 : 1], Max error [0.0479]
CMU, x64, vs2017, avx, 4 cores
Compressed 2534 clips in 00h 12m 39.22s
Compressed 67.04 MB, Elapsed 00h 46m 33.27s, Ratio [21.32 : 1], Max error [0.0479]
CMU, x64, vs2017, avx, 3 cores
Compressed 2534 clips in 00h 15m 55.93s
Compressed 67.04 MB, Elapsed 00h 44m 03.44s, Ratio [21.32 : 1], Max error [0.0479]

CMU, x86, gcc5, avx, 3 cores
No wall clock time
Compressed 67.04 MB, Elapsed 01h 40m 31.27s, Ratio [21.32 : 1], Max error [0.0462]
CMU, x64, gcc5, avx, 3 cores
No wall clock time
Compressed 67.04 MB, Elapsed 01h 52m 53.10s, Ratio [21.32 : 1], Max error [0.0479]

CMU, x86, clang5, avx, 3 cores
No wall clock time
Compressed 67.04 MB, Elapsed 00h 47m 51.94s, Ratio [21.32 : 1], Max error [0.0479]
CMU, x64, clang5, avx, 3 cores
No wall clock time
Compressed 67.04 MB, Elapsed 00h 35m 11.64s, Ratio [21.32 : 1], Max error [0.0479]
